src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -30712,20 +30712,567 @@ public interface I1
 
     static I1() {}
 }
+interface I2
+{
+    I2();
+}
+
 ";
             var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
                                                  parseOptions: TestOptions.Regular);
 
-            // From https://github.com/dotnet/csharplang/blob/master/meetings/2017/LDM-2017-05-17.md:
-            // We allow initializers for static fields, but not an explicit static constructor. Can think about adding later.
             compilation1.VerifyDiagnostics(
-                // (4,5): error CS0526: Interfaces cannot contain constructors
+                // (4,5): error CS0526: Interfaces cannot contain instance constructors
                 //     I1(){}
                 Diagnostic(ErrorCode.ERR_InterfacesCantContainConstructors, "I1").WithLocation(4, 5),
-                // (6,12): error CS0526: Interfaces cannot contain constructors
+                // (10,5): error CS0501: 'I2.I2()' must declare a body because it is not marked abstract, extern, or partial
+                //     I2();
+                Diagnostic(ErrorCode.ERR_ConcreteMissingBody, "I2").WithArguments("I2.I2()").WithLocation(10, 5),
+                // (10,5): error CS0526: Interfaces cannot contain instance constructors
+                //     I2();
+                Diagnostic(ErrorCode.ERR_InterfacesCantContainConstructors, "I2").WithLocation(10, 5)
+                );
+        }
+
+        [Fact]
+        public void Constructors_02()
+        {
+            var source1 =
+@"
+interface I1
+{
+    static I1() {}
+}
+interface I2
+{
+    static I2() => throw null;
+}
+interface I3
+{
+    I3() {}
+}
+interface I4
+{
+    I4() => throw null;
+}
+interface I5
+{
+    I5();
+}
+interface I6
+{
+    extern static I6();
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.Regular7_3);
+
+            compilation1.VerifyDiagnostics(
+                // (4,12): error CS8652: The feature 'default interface implementation' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
```

Comment:
Nit: is there any way to suppress this error? Otherwise, when we eventually ship C# 8.0, this error will need to be deleted from this test and who knows how many others.

Reply:
> Nit: is there any way to suppress this error? Otherwise, when we eventually ship C# 8.0, this error will need to be deleted from this test and who knows how many others.

It is the purpose of the test to verify presence of this diagnostics. The error will never go away for the scenario, the wording might change though, but that is due to the "Preview" mode ending.

---
In reply to: [260427140](https://github.com/dotnet/roslyn/pull/33625#discussion_r260427140) [](ancestors = 260427140)